sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1057,56 +1057,90 @@ private static void validateMethodAnnotations(
       methodNameToAllMethodMap.put(method, method);
     }
 
-    List<InconsistentlyIgnoredGetters> incompletelyIgnoredGetters = new ArrayList<>();
-    List<IgnoredSetter> ignoredSetters = new ArrayList<>();
+    // Verify that there is no getter with a mixed @JsonIgnore annotation.
+    validateGettersNoMixedAnnotations(
+        methodNameToAllMethodMap, descriptors, AnnotationPredicate.JSON_IGNORE);
 
+    // Verify that there is no getter with a mixed @Default annotation.
+    validateGettersNoMixedAnnotations(
+        methodNameToAllMethodMap, descriptors, AnnotationPredicate.DEFAULT_VALUE);
+
+    // Verify that no setter has @JsonIgnore.
+    validateSettersNoAnnotations(
+        methodNameToAllMethodMap, descriptors, AnnotationPredicate.JSON_IGNORE);
+
+    // Verify that no setter has @Default.
+    validateSettersNoAnnotations(
+        methodNameToAllMethodMap, descriptors, AnnotationPredicate.DEFAULT_VALUE);
+  }
+
+  /**
+   * Validates that getters don't have mixed annotation.
+   */
+  private static void validateGettersNoMixedAnnotations(
+      SortedSetMultimap<Method, Method> methodNameToAllMethodMap,
+      List<PropertyDescriptor> descriptors,
+      AnnotationPredicate annotationPredicate) {
+    List<InconsistentlyAnnotatedGetters> incompletelyAnnotatedGetters = new ArrayList<>();
     for (PropertyDescriptor descriptor : descriptors) {
       if (descriptor.getReadMethod() == null
-          || descriptor.getWriteMethod() == null
-          || IGNORED_METHODS.contains(descriptor.getReadMethod())
-          || IGNORED_METHODS.contains(descriptor.getWriteMethod())) {
+          || IGNORED_METHODS.contains(descriptor.getReadMethod())) {
         continue;
       }
-      // Verify that there is no getter with a mixed @JsonIgnore annotation and verify
-      // that no setter has @JsonIgnore.
       SortedSet<Method> getters = methodNameToAllMethodMap.get(descriptor.getReadMethod());
-      SortedSet<Method> gettersWithJsonIgnore = Sets.filter(getters, JsonIgnorePredicate.INSTANCE);
+      SortedSet<Method> gettersWithTheAnnotation = Sets.filter(getters, annotationPredicate);
```

Comment:
In the case of @Default, this is missing the logic to make sure that the two distinct annotation instances are not equal.

So this will currently say that @Default.String("a") and @Default.String("b") are equivalent which is incorrect.

Conveniently you can use Annotation#equals as it does the right thing:
https://docs.oracle.com/javase/7/docs/api/java/lang/annotation/Annotation.html#equals(java.lang.Object)